pkgs/by-name/an/angular-language-server-bin/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  nodejs,
+  makeBinaryWrapper,
+  runCommand,
+  angular-language-server-bin,
+  writeShellApplication,
+  curl,
+  common-updater-scripts,
+  jq,
+  unzip,
+}:
+
+let
+  owner = "angular";
+  repo = "vscode-ng-language-service";
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "angular-language-server-bin";
+  version = "18.2.0";
+  src = fetchurl {
+    name = "${finalAttrs.pname}-${finalAttrs.version}.zip";
+    url = "https://github.com/${owner}/${repo}/releases/download/v${finalAttrs.version}/ng-template.vsix";
+    hash = "sha256-rl04nqSSBMjZfPW8Y+UtFLFLDFd5FSxJs3S937mhDWE=";
+  };
+
+  nativeBuildInputs = [
+    unzip
+    makeBinaryWrapper
+  ];
+
+  buildInputs = [ nodejs ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 server/bin/ngserver $out/bin/ngserver
+    install -Dm755 server/index.js $out/index.js
+    cp -r node_modules $out/node_modules
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    patchShebangs $out/bin/ngserver $out/index.js $out/node_modules
+    wrapProgram $out/bin/ngserver \
+      --prefix PATH : ${lib.makeBinPath [ nodejs ]} \
+      --add-flags "--tsProbeLocations $out/node_modules --ngProbeLocations $out/node_modules"
+  '';
+
+  passthru = {
+    tests = {
+      start-ok = runCommand "${finalAttrs.pname}-test" { } ''
+        ${angular-language-server-bin}/bin/ngserver --stdio --help &> $out
```

Comment:
We can use lib.getExe here and grep in one go

Reply:
Added lib.getExe.